lib/kafka/consumer.rb

```diff
@@ -218,6 +227,9 @@ def each_message(min_bytes: 1, max_bytes: 10485760, max_wait_time: 1, automatica
 
       consumer_loop do
         batches = fetch_batches
+        unless @interceptors.nil?
+          batches = @interceptors.on_consume(batches)
```

Comment:
>the interceptor should only receive valid messages.

the interceptor could do something with invalid messages

Reply:
I could intercept each batch instead of the list of batches. It looks like that is what the [java lib](https://github.com/apache/kafka/commit/d5b43b19bb06e9cdc606312c8bcf87ed267daf44#diff-cc31c241ae7360e160b951db1ef8135bR57) does